sys/memarray/memarray.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2018 Tobias Heider <heidert@nm.ifi.lmu.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+#include <string.h>
+#include "memarray.h"
+
+#define ENABLE_DEBUG    (0)
+#include "debug.h"
+
+void memarray_init(memarray_t *mem, void *data, size_t size, size_t num)
+{
+    assert((mem != NULL) && (data != NULL) && (size >= sizeof(void *)));
```

Comment:
In the same part of having assert for the API I also saw that there could be an assert for `num != 0`  as its a case that is not handled right now.

Reply:
True, thanks for the hint